### ISSUE TITLE
Reindex OpenSearch to address link checker gap

### ISSUE THREAD

**Issue by sfisher** [CDLUC3/ezid#729]
There was a gap between our initial indexing of documents for OpenSearch and some fields that that link checker updated without accurately updating the update time timestamp.  While we reindexed things that had been updated, the link checker is above the law when it makes changes and doesn't reflect updated records.

Because of this, there may be some records with

https://github.com/CDLUC3/ezid/issues/618 outdated "linkIsBroken" and "hasIssues" from the search table.  We can fix this by just reindexing everything and overwriting the old info.  This should be a fairly hands-off process aside from restarting or correcting problems if the reindexing dies in the middle.

Going forward, we shouldn't see this issue since the linkChecker in the current code writes to both data stores.

**Comment by sfisher** [comment]
The task to reindex worked OK, but running as fast as possible (batching 100 at a time and going through all) placed additional load on the servers which caused Nagios alerts because it degraded search time after an hour or two of updates.  We were able to see the same pattern on both production and stage servers.

There are a couple of possible solutions to this issue:

1. Add more wait time between updates.  I tested this solution on Stage last week and it ran all week without causing issues.  The pattern was to do a batch (100) and then sleep 5 seconds, repeat.  This worked OK without causing undue load or slowness.

OR

2. After thinking and attending the OpenSearch conference we figured out a couple of elements and techniques and it sounds like the plan is:
  a. Update code to ensure that the `ezidapp_searchidentifier` is correctly updated any time the record is updated.  This allows us to reliably update only records that have changed since a certain date in order to do incremental updates of only records that require them after earlier updates.
  b. See if we can set an alias for our index in Opensearch and be sure that alias works.
  c. Point our in-use index to use the alias rather than the index name directly.
  d. Create an entirely new index offline (ie the index is not in production use).
  e. On index completion do another round of updates only on items updated after indexing started.
  f.  Swap the new index for the old one by pointing the alias at the new index.
  g. Another super-mini update for only like the last few hours or day period for those updates.
  h. Run the script that removes items deleted from the database so they are also removed from OpenSearch, while it scans a lot of things it doesn't make that many changes to the index since we don't delete that many IDs.
